### PR TITLE
[config] Map into config.json

### PR DESCRIPTION
**Pull request by jaylees14**
This generates `config.json` for each service. 

Couple of things were found whilst doing this:
- Computing which services can be accessed also needs to take into account any structs within that service
- Go looks service keys up using `CamelCase` names, but then the URL needs `kebab-case` - are we okay with these being different?
- I don't like the user of `.get`, but we're pretty certain it's going to be in there - what should I do instead? Throw?
- None of our examples in ProjectBuilder or E2E test actually include a foreign key.
  - Rather than add one and have to deal with Go changes I just created a new test

### CODE REVIEW

File: src/main/scala/temple/builder/project/ProjectBuilder.scala [hunk 1]
```diff
@@ -114,9 +115,18 @@ object ProjectBuilder {
       case (name, service, port) =>
         val serviceRoot =
           ServerBuilder.buildServiceRoot(name, service, port.service, endpoints(service), detail, usesAuth)
-        service.lookupMetadata[ServiceLanguage].getOrElse(ProjectConfig.defaultLanguage) match {
+        val serverFiles = service.lookupMetadata[ServiceLanguage].getOrElse(ProjectConfig.defaultLanguage) match {
           case ServiceLanguage.Go => GoServiceGenerator.generate(serviceRoot)
         }
+
+        val serviceComms = serviceRoot.comms.map { service =>
+          val (_, _, ports) = templefile.providedServicesWithPorts.find { _._1 == service }.get
```
Comment (c-harding) [inline src/main/scala/temple/builder/project/ProjectBuilder.scala:123]: I see, maybe worth extracting this to a function in `Templefile`? But it’s only unsafe if the service isn’t really in the templefile

**Review by c-harding — APPROVED** [review]
looks good to me. is the kebabbing of file names going to be problematic?

File: src/main/scala/temple/builder/project/ProjectBuilder.scala [hunk 2]
```diff
@@ -114,9 +115,18 @@ object ProjectBuilder {
...
+          service -> s"http://${kebabCase(service)}:${ports.service}"
+        }.toMap
+
+        val configFileContents =
+          ServerConfigGenerator.generate(serviceRoot.kebabName, serviceRoot.datastore, serviceComms, port)
+        serverFiles + (File(serviceRoot.kebabName, "config.json") -> configFileContents)
```
Comment (Lewiky) [inline src/main/scala/temple/builder/project/ProjectBuilder.scala]:
```suggestion
        val configFileContents =
          ServerConfigGenerator.generate(serviceRoot.kebabName, serviceRoot.datastore, serviceComms, port)
          
        serverFiles + (File(serviceRoot.kebabName, "config.json") -> configFileContents)
```

**Review by Lewiky — APPROVED** [review]
LGTM

File: src/e2e/resources/simple-temple-expected/booking/config.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+  "user" : "postgres",
+  "dbName" : "postgres",
+  "host" : "booking-db",
+  "sslMode" : "disable",
+  "services" : {
+    
+  },
```
Comment (willjvsmith) [inline src/e2e/resources/simple-temple-expected/booking/config.json:8]: I guess it doesn't matter, empty block

**Pull request merged**